### PR TITLE
add GetWorkSpaceSize to base arg

### PR DESCRIPTION
**Pull request by shaojiewang**
Add GetWorkSpaceSize to base arg and make an example on convnd_bwd_we…ight

### CODE REVIEW

File: include/ck/tensor_operation/gpu/device/device_convnd_backward_weight_xdl_c_shuffle_nhwc_kyxc_nhwk.hpp [hunk 1]
```diff
@@ -1175,6 +1175,67 @@ struct DeviceConvndBwdWeightXdl_C_Shuffle_Input_N_Hi_Wi_C_Weight_K_Y_X_C_Output_
...
+        size_t WorkSpaceSize = 0;
+        if constexpr(std::is_same<InDataType, ck::bhalf_t>::value)
+        {
+            WorkSpaceSize =
+                arg.Conv_K_ * arg.Conv_C_ * arg.filter_spatial_lengths_[0] * sizeof(float);
+        }
+        else
+        {
+            WorkSpaceSize = arg.Conv_K_ * 0;
+        }
```
Comment (asroy) [inline include/ck/tensor_operation/gpu/device/device_convnd_backward_weight_xdl_c_shuffle_nhwc_kyxc_nhwk.hpp:1191]: Why not just remove?

File: example/20_convnd_bwd_weight_xdl/convnd_bwd_weight_xdl.cpp [hunk 2]
```diff
@@ -296,15 +296,53 @@ int main(int argc, char* argv[])
                                   OutElementOp{},
                                   split_k);
 
-    if(!conv->IsSupportedArgument(argument.get()))
+    // alloc work space
+    size_t bwd_weight_workspace_size = conv->GetWorkSpaceSize(argument.get());
+    float ave_time                   = 0.f;
+    if(bwd_weight_workspace_size > 0)
```
Comment (asroy) [inline example/20_convnd_bwd_weight_xdl/convnd_bwd_weight_xdl.cpp:302]: I think the predicate should be "whether the datatype is bf16 or not". ```bwd_weight_workspace_size``` is a result of the predicate, not the predicate itself.

File: include/ck/tensor_operation/gpu/device/device_convnd_backward_weight_xdl_c_shuffle_nhwc_kyxc_nhwk.hpp (hunk 1 above)
Comment (shaojiewang) [reply include/ck/tensor_operation/gpu/device/device_convnd_backward_weight_xdl_c_shuffle_nhwc_kyxc_nhwk.hpp:1191]: yes, removed.

File: example/20_convnd_bwd_weight_xdl/convnd_bwd_weight_xdl.cpp (hunk 2 above)
Comment (shaojiewang) [reply example/20_convnd_bwd_weight_xdl/convnd_bwd_weight_xdl.cpp]: I agree. Changed.

**Pull request merged**